boa_engine/src/builtins/temporal/plain_date_time/mod.rs

```diff
@@ -0,0 +1,148 @@
+//! Boa's implementation of the ECMAScript `Temporal.PlainDateTime` builtin object.
+#![allow(dead_code, unused_variables)]
+
+use crate::{
+    builtins::{BuiltInBuilder, BuiltInConstructor, BuiltInObject, IntrinsicObject},
+    context::intrinsics::{Intrinsics, StandardConstructor, StandardConstructors},
+    property::Attribute,
+    realm::Realm,
+    string::common::StaticJsStrings,
+    Context, JsNativeError, JsObject, JsResult, JsString, JsSymbol, JsValue,
+};
+use boa_profiler::Profiler;
+
+use self::iso::IsoDateTimeRecord;
+
+pub(crate) mod iso;
+
+/// The `Temporal.PlainDateTime` object.
+#[derive(Debug, Clone)]
+pub struct PlainDateTime {
+    pub(crate) inner: IsoDateTimeRecord,
+    pub(crate) calendar: JsValue,
```

Comment:
Yes, the `calendar` field is typically either an object representing a calendar or a string. It's a little weird because the `Instance` portion of the spec identifies the value as an object. However, `CreateTemporalDateTime` lists the value as: "calendar (a String or Object)".

When a method is called using the calendar, it typically defaults to the `Calendar-x` method, which essentially looks up the calendar and calls the method.